src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -249,4 +276,255 @@ public void equalsReturnsFalseForNotEqualBeings(
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+
+        assertThat(initialBuildStatus).isLessThanOrEqualTo(finalBuildStatus);
+    }
+
+    @Test
+    public void pawnDoNotBuildsHouses() {
+        // Arrange
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final IPathFinder pathFinder = Mockito.mock(IPathFinder.class);
+        final Position position = new Position(0, 0);
+        final Position farAway = new Position(50, 50);
+        final IStructure structure = new House(farAway);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(
+            position,
+            position,
+            Mockito.mock(Random.class),
+            pathFinder,
+            mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+        // Value did not change because Pawn was not close enough
+        assertThat(initialBuildStatus).isEqualTo(finalBuildStatus);
+    }
+
+    @Test
+    public void doesPawnWalkToNearestUnBuiltBuilding() {
+        // Arrange
+        final Position positionA = new Position(10, 0);
+        final Position startPosition = new Position(20, 0);
+        final Position positionB = new Position(30, 0);
+
+        final Position correctDestination = new Position(11, 0);
+
+        final World world = new World(50, 0);
+        final IPathFinder pathFinder = new AstarPathFinder(world);
+
+        world.createStructure(positionA);
+        world.createStructure(positionB);
+
+        for (int i = 0; i < 10; i++) {
+            world.getStructureAt(positionB).get().deliverItem(new Rock());
+            world.getStructureAt(positionB).get().deliverItem(new Log());
+
+        }
+
+        final Pawn testPawn = new Pawn(
+            startPosition,
+            new Position(),
+            new Random(),
+            pathFinder,
+            world);
+
+        // Act
+        testPawn.update();
+
+        // Assert
+        assertThat(testPawn.getFinalDestination().get()).isEqualTo(correctDestination);
+    }
+
+    @Test
+    public void allStructuresCompleted() {
+        // Arrange
+        final Position positionA = new Position(10, 0);
+        final Position startPosition = new Position(20, 0);
+        final Position positionB = new Position(30, 0);
+
+        final Position correctDestination = new Position(0, 0);
+
+        final World world = new World(50, 0);
+        final IPathFinder pathFinder = new AstarPathFinder(world);
+
+        final Random mockRandom = Mockito.mock(Random.class);
+        when(mockRandom.nextInt(anyInt())).thenReturn(0);
+        when(mockRandom.nextInt(anyInt())).thenReturn(0);
+
+        world.createStructure(positionA);
+        world.createStructure(positionB);
+
+        for (int i = 0; i < 10; i++) {
```

Comment:
How does the test know that 10 of each item is required?

Looks like it knows it because it knows about the implementation details of house, and it knows that house internally says that it needs 10 of these items. 

This is incorrect, as the client code (currently this test) should have a real way of knowing for sure how many items are required. I believe this is possible using the current getters, but if it's not then this is a sign that you're missing a part of the required API.

Reply:
The reason was to make sure no tests would run forever, which could be the case if it would check what would be needed by the house. Needing certain items to be constructed could be an requirement on a specific house, if so it is not out of place.